src/awkward/_errors.py

```diff
@@ -18,6 +19,22 @@
 
 
 E = TypeVar("E", bound=Exception)
+T = TypeVar("T")
+S = TypeVar("S")
+P = ParamSpec("P")
+
+
+class WeakMethodProxy:
+    """A proxy for a method of a weakly referenced object"""
+
+    def __init__(self, method):
+        self._this = weak_ref(method.__self__)
+        self._impl = method.__func__
+
+    def __call__(self, *args, **kwargs):
+        this = self._this()
+        method = self._impl.__get__(this, type(this))
+        return method(*args, **kwargs)
```

Comment:
Okay, that's good to know. This error-handling context has changed several times, but it was originally introduced so that kernels can more fully occupy the device (#1321).

Here's what

```python
a = np.random.uniform(5, 10, 1000000)
b = np.random.uniform(10, 20, 1000000)
c = np.random.uniform(-0.1, 0.1, 1000000)
```

looks like, computing

```python
(-b + np.sqrt(b**2 - 4*a*c)) / (2*a)
```

![download](https://github.com/scikit-hep/awkward/assets/1852447/9acd3a75-37da-45e9-8773-a4a1221d8728)

It's pretty hard to get one to run immediately after the next because they're so quick compared to the time for Python to execute an instruction.